src/http/gateway/resources/gateway.js

```diff
@@ -56,10 +56,11 @@ module.exports = {
     // so we convert /ipns/ to /ipfs/ before passing it to the resolver ¯\_(ツ)_/¯
     // This could be removed if a solution proposed in
     //  https://github.com/ipfs/js-ipfs-http-response/issues/22 lands upstream
-    const ipfsPath = decodeURI(path.startsWith('/ipns/')
+    let ipfsPath = decodeURI(path.startsWith('/ipns/')
       ? await ipfs.name.resolve(path, { recursive: true })
       : path)
 
+    let directory = false
```

Comment:
I went with explicit flag to make it easier to review code paths related to directories (with and without `index.html`)